### PR TITLE
Update eth-scan to 3.1.2

### PR DESCRIPTION
**Pull request by Mrtenz**

### CODE REVIEW

File: src/services/Store/BalanceService.tsx [hunk 1]
```diff
@@ -20,18 +19,19 @@ const getAssetAddresses = (assets: Asset[] = []): (string | undefined)[] => {
   return assets.map((a) => a.contractAddress).filter((a) => a);
 };
 
-export const convertBNToBigNumberJS = (bn: EthScanBN): BN => {
-  return new BN(bn._hex);
+export const convertBNToBigNumberJS = (bn: bigint): BN => {
+  console.log(bn);
```
Comment (FrederikBolding) [inline src/services/Store/BalanceService.tsx:23]: Please remove 😄

File: src/services/Store/StoreProvider.tsx [hunk 2]
```diff
@@ -134,6 +135,25 @@ export interface State {
 }
 export const StoreContext = createContext({} as State);
 
+// @todo: Update unlock-scan to use native bigints like eth-scan
```
Comment (FrederikBolding) [inline src/services/Store/StoreProvider.tsx:138]: Can we do this now such that we don't have to duplicate converting logic?

Comment (FrederikBolding) [reply src/services/Store/StoreProvider.tsx:138]: If not, please move this to a helper and test.

**Review by FrederikBolding — CHANGES_REQUESTED** [review]

File: src/services/Store/BalanceService.tsx [hunk 3]
```diff
@@ -100,23 +99,22 @@ const tokenBalanceFetchWrapper = async (
   address: string,
   contractList: string[],
   options: any
-): Promise<EthScanBalanceMap<EthScanBN>> => {
+): Promise<BalanceMap<bigint>> => {
   try {
-    const tokenBalanceMap = await getTokensBalance(provider, address, contractList, options);
-    return tokenBalanceMap;
+    return getTokensBalance(provider, address, contractList, options);
```
Comment (GuillaumeRx) [inline src/services/Store/BalanceService.tsx:104]: Missing to await ?

File: src/services/Store/BalanceService.tsx [hunk 4]
```diff
@@ -86,12 +86,11 @@ const etherBalanceFetchWrapper = async (
   provider: ProviderLike,
   address: string,
   options: any
-): Promise<EthScanBalanceMap<EthScanBN>> => {
+): Promise<BalanceMap<bigint>> => {
   try {
-    const balanceMap = await getEtherBalances(provider, [address], options);
-    return balanceMap;
+    return getEtherBalances(provider, [address], options);
```
Comment (GuillaumeRx) [inline src/services/Store/BalanceService.tsx:91]: Missing to await ?

File: src/services/Store/BalanceService.tsx (hunk 1 above)
Comment (Mrtenz) [reply src/services/Store/BalanceService.tsx]: Fixed

**Comment by FrederikBolding** [comment]
Still WIP according to @Mrtenz

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**